engine/src/witness/btc.rs

```diff
@@ -274,7 +318,7 @@ impl VoterApi<BitcoinEgressWitnessingES> for BitcoinEgressWitnessingVoter {
 			tracing::info!("Witnesses from BTCE: {:?}", witnesses);
 		}
 
-		Ok(Some(witnesses))
+		Ok(Some((witnesses, response_block_hash.map(|hash| H256::from_slice(hash.as_ref())))))
```

Comment:
Previously we didn't transfer the hash between statechain <-> engine at all! Still a bit inconvenient that we have to do this now...